internal/db/storage/blob/s3/option.go

```diff
@@ -30,6 +33,34 @@ type Option func(c *client) error
 var (
 	defaultOpts = []Option{
 		WithErrGroup(errgroup.Get()),
+
+		func(c *client) error {
+			c.readerFunc = func(key string) (reader.Reader, error) {
+				return reader.New(
+					reader.WithErrGroup(c.eg),
+					reader.WithService(c.service),
+					reader.WithBucket(c.bucket),
+					reader.WithKey(key),
+					reader.WithMaxChunkSize(c.maxChunkSize),
+					reader.WithBackoff(c.readerBackoffEnabled),
+					reader.WithBackoffOpts(c.readerBackoffOpts...),
+				)
+			}
+			return nil
+		},
+
+		func(c *client) error {
+			c.writerFunc = func(key string) writer.Writer {
+				return writer.New(
+					writer.WithErrGroup(c.eg),
+					writer.WithService(c.service),
+					writer.WithBucket(c.bucket),
+					writer.WithKey(key),
+					writer.WithMaxPartSize(c.maxPartSize),
+				)
+			}
+			return nil
+		},
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blueviolet.svg?style=flat-square)
how do you test these functions?